src/PHPStan/Faker/Reflection/FakerMethodsClassReflectionExtension.php

```diff
@@ -74,16 +74,86 @@ public function __construct()
 			'randomElements' => [new StringType(), false, false, true, [$this->ap('array'), $this->ip('count')], false], //@method array randomElements(array $array = array('a', 'b', 'c'), $count = 1)
 			'shuffleArray' => [new StringType(), false, false, true, [$this->ap('array')], false], //@method array shuffleArray(array $array = array())
 
-			'boolean' => [new TrueOrfalseBooleanType(), false, false, true, [$this->ip('chanceOfGettingTrue')], false], //@method boolean boolean($chanceOfGettingTrue = 50)
+			'boolean' => [new TrueOrFalseBooleanType(), false, false, true, [$this->ip('chanceOfGettingTrue')], false], //@method boolean boolean($chanceOfGettingTrue = 50)
 
 			'randomFloat' => [new FloatType(), false, false, true, [$this->ip('nbMaxDecimals'), $this->ip('min'), $this->ip('max')], false], //@method float randomFloat($nbMaxDecimals = null, $min = 0, $max = null)
 		];
 	}
 
+	private function sp(string $name): ParameterReflection
+	{
+		return $this->createParameterInstance(new StringType(), $name, true, false, false);
+	}
+
+	private function createParameterInstance(Type $type, string $name, bool $optional, bool $passedByReference, bool $variadic): ParameterReflection
+	{
+		return new class($type, $name, $optional, $passedByReference, $variadic) implements ParameterReflection
+		{
+			/**
+			 * @var mixed
+			 */
+			private $type, $name, $optional, $passedByReference, $variadic;
+
+			public function __construct(Type $type, string $name, bool $optional, bool $passedByReference, bool $variadic)
+			{
+				$this->type = $type;
+				$this->name = $name;
+				$this->optional = $optional;
+				$this->passedByReference = $passedByReference;
+				$this->variadic = $variadic;
+			}
+
+			public function getName(): string
+			{
+				return $this->name;
+			}
+
+			public function isOptional(): bool
+			{
+				return $this->optional;
+			}
+
+			public function getType(): Type
+			{
+				return $this->type;
+			}
+
+			public function isPassedByReference(): bool
+			{
+				return $this->passedByReference;
+			}
+
+			public function isVariadic(): bool
+			{
+				return $this->variadic;
+			}
+		};
+	}
+
+	private function bp(string $name): ParameterReflection
+	{
+		return $this->createParameterInstance(new TrueOrFalseBooleanType(), $name, true, false, false);
+	}
+
+	private function ip(string $name): ParameterReflection
+	{
+		return $this->createParameterInstance(new IntegerType(), $name, true, false, false);
+	}
+
+	private function ap(string $name): ParameterReflection
+	{
+		return $this->createParameterInstance(new ArrayType(new MixedType(), new MixedType(), false), $name, true, false, false);
+	}
+
+	private function fp(string $name): ParameterReflection
+	{
+		return $this->createParameterInstance(new FloatType(), $name, true, false, false);
+	}
+
```

Comment:
Why is the position of this block changed?

Reply:
Sorry, it's mistake. I return back.